src/Perpetuum/Services/EventServices/EventProcessors/DayTimeEventProcessor.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using Perpetuum.ExportedTypes;
+using Perpetuum.Services.Daytime;
+using Perpetuum.Services.EventServices.EventMessages;
+using Perpetuum.Zones;
+using Perpetuum.Zones.Effects.ZoneEffects;
+
+namespace Perpetuum.Services.EventServices.EventProcessors
+{
+    public class DayTimeEventProcessor : EventProcessor<EventMessage>
+    {
+        private IZone _zone;
+        private readonly Lazy<ZoneEffect> _dayEffect;
+        private readonly Lazy<ZoneEffect> _nightEffect;
+        public DayTimeEventProcessor(IZone zone)
+        {
+            _zone = zone;
+            _dayEffect = new Lazy<ZoneEffect>(GetDayEffect);
+            _nightEffect = new Lazy<ZoneEffect>(GetNightEffect);
+        }
+
+        private ZoneEffect GetDayEffect()
+        {
+            return new ZoneEffect(_zone.Id, EffectType.effect_daytime_day, true);
+        }
+
+        private ZoneEffect GetNightEffect()
+        {
+            return new ZoneEffect(_zone.Id, EffectType.effect_daytime_night, true);
+        }
+
+        private static readonly int _sunrise = 150;
+        private static readonly int _sunset = 750;
+
+        private bool IsDay(DayTimeMessage msg)
+        {
+            return msg.TimeInfo.NLT < 500 && msg.TimeInfo.NLT > 300;
+        }
+
+        private bool IsNight(DayTimeMessage msg)
+        {
+            return msg.TimeInfo.NLT < 50 || msg.TimeInfo.NLT > 850;
+        }
+
+        public override void OnNext(EventMessage value)
+        {
+            if (value is DayTimeMessage msg)
+            {
+                if (IsNight(msg))
+                {
+                    _zone.ZoneEffectHandler.AddEffect(_nightEffect.Value);
+                }
+                else
+                {
+                    _zone.ZoneEffectHandler.RemoveEffect(_nightEffect.Value);
+                }
+
+                if (IsDay(msg))
+                {
+                    _zone.ZoneEffectHandler.AddEffect(_dayEffect.Value);
+                }
+                else
+                {
+                    _zone.ZoneEffectHandler.RemoveEffect(_dayEffect.Value);
+                }
```

Comment:
Pretty sure this can be contracted to a single branch;

```
if (IsDay(msg))
{
  _zone.ZoneEffectHandler.RemoveEffect(_nightEffect.Value);
  _zone.ZoneEffectHandler.AddEffect(_dayEffect.Value);
}
else
{
  _zone.ZoneEffectHandler.RemoveEffect(_dayEffect.Value);
  _zone.ZoneEffectHandler.AddEffect(_nightEffect.Value);
}
```

Reply:
The union of "day" and "night" time-domains do not necessarily cover the full daytime.  I wanted to leave open the possibility for there to be periods where no bonus is applied.  
Chances are I should just make this enumerable and configurable with a DB table of params, but I didn't want to go overboard on the first pass of a relatively non-consequential feature.

Reply:
I get what you're saying, but as it stands you're forcing 2 decision branches which involves all the work the processor does to try and read, check, and predict that.

If you want to leave it as is, fair enough, but given that we don't currently have anything other than Day or Night it would be more performant to only have a single decision branch. Up to you.